Prevent push notification from duplicating on android

ramza1 opened the pull request:
When a push notification is received it is being duplicated on the android devise and on parse push dashboard it says just one was sent

Commit:
upgraded Parse Android SDK to 1.7.1. No more PushService.

Commit:
rename initialize() method to initializeParse because we don't want to
override CordovaPlugin.initialize()

Commit:
remove support for non-GCM devices to eliminate the crash on screen
unlock (due to ParseBroadcastReceiver being invoked by
android.intent.action.USER_PRESENT before Parse.initialize is called)

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
removed old comments from plugin.xml

Commit:
replaced initialize() for register().  allows registration of a
Javascript ecb to be called upon receiving a push notification.
Restructured and renamed plugin to be more descriptive: ParsePushPlugin.

Commit:
updated docs

Commit:
update docs

Commit:
upgraded Parse android sdk to 1.8.1

Commit:
do away with string-based ecb in favor of Parse.Events for onReceive
event handling

Commit:
made appId/clientKey optional in the js register() function. Those opts
are now use for quickstart testing only. Updated docs to show user how
to register device correctly using MainApplication.java

Commit:
update docs

Commit:
prevent flooding notification tray during multiple notifications by
retaining only the last notif with the same title field.

Commit:
makes docs clearer

Commit:
add a count to notification tray item to show how many PNs have arrived
for each "title"

Commit:
only create entry in notification tray if application is not in
foreground.

Commit:
opening a notification now brings the main activity to front if it's
running in background (does not do unnecessary reload)

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
added support for openPN event. added support for urlHash field in PN
that allows app to start on a specific view when a PN is open. urlHash +
openPN event also allow paused app to be resumed and navigate to a
specific view

Commit:
comment out some logs

Commit:
update plugin.xml

benjie commented:
Thanks for the pull request but I no longer maintain this repository so I won't be merging - sorry!

Commit:
Android Support Library Added
Modified code to handle closed application case
Mentioned need of Parse JS SDK

saurabhgupta050890 committed:
Update README.md

taivo committed:
Merge pull request #1 from saurabhgupta050890/master

Build failure on Cordova 5.0

Commit:
To enable notification sound when app is in background

Commit:
upgraded Parse android to v1.10.1, bolts to 1.2.1, and
android-support-v4 to latest as of 9/2015

Commit:
ensure Parse js is loaded before declaring plugin

Commit:
do away with webView.sendJavascript and use cordova.exec bridge to call
the javascript pn handler. This way we don't have to set
Content-Security-Policy to "script-src 'unsafe-eval'"

Commit:
do away with javascript based device registration. It has to be done in
Android any way.

Commit:
fixed a bug that prevents customEvents from firing. Also made 'openPN',
'receivePN' a member of ParsePushPlugin object so those event names can
be changed by developer.

Commit:
version bump

abeninski committed:
Fixed app crash when notification is received and app is closed

taivo committed:
Merge pull request #3 from abeninski/master

Fixed app crash when notification is received and app is closed

taivo committed:
Merge pull request #2 from aegisrunestone/patch-1

To enable notification sound when app is in background

Commit:
removed underscore dependency. copied event handling code into this
plugin so no longer require Parse.Events

Commit:
push icon color now customizable for Android Lollipop

ilesinge committed:
Fix crash on getSubscriptions when channel field is empty on parse

taivo committed:
Merge pull request #13 from ilesinge/master

Fix crash on getSubscriptions when channel field is empty on parse

taivo committed:
update Parse framework and libs

taivo committed:
iOS support!

taivo committed:
updated docs

taivo committed:
format the pnPayload to be consistent with Android version of this
plugin

dirkpostma committed:
Solve issue 23

https://github.com/taivo/parse-push-plugin/issues/23

taivo committed:
fixed iOS cold start issue #24

taivo committed:
Merge pull request #25 from dirkpostma/patch-1

Solve issue 23

taivo committed:
retain push notifications in a queue, flush queue when registerCallback
occurs

taivo committed:
Merge branch 'master' of https://github.com/taivo/parse-push-plugin

etabard committed:
Track open notifications

taivo committed:
Merge pull request #29 from etabard/track-open

Track open notifications

taivo committed:
should fix issue #31, crash when receiving notif in iOS

campers committed:
Updated for parse android 1.13.0 on the open source server

taivo committed:
Merge pull request #40 from campers/android-update

Updated for parse android 1.13.0 on the open source server

taivo committed:
update ios libs to work with oss parse-server

taivo committed:
remove unnecessary parse libs

taivo committed:
iOS now works for oss parse-server

taivo committed:
updated docs: remove references to setup for Android without GCM

taivo committed:
update docs to mention how to push custom subevents

taivo committed:
update docs and bump version

taivo committed:
update docs

taivo committed:
docs

taivo committed:
update docs for Android setup. removed GET_ACCOUNTS permission on Android

taivo committed:
docs

taivo committed:
docs

taivo committed:
docs

taivo committed:
docs

taivo committed:
include installation instruction for old parse-com branch

taivo committed:
simplify iOS setup. No longer need to modify AppDelegate.m

taivo committed:
docs

taivo committed:
simplified android setup. no longer need to declare own MainApplication and touch AndroidManifest

taivo committed:
update docs

taivo committed:
update docs

taivo committed:
update docs

taivo committed:
update docs

taivo committed:
prepping for npm publish

taivo committed:
added package.json

taivo committed:
package.json

taivo committed:
docs

taivo committed:
added .gitignore

taivo committed:
fixed cold start openPN issue #24 for Android

taivo committed:
allow advanced users to set _coldStartDelayMs

taivo committed:
android installation will now warn user if android:name is already set

taivo committed:
fixed iOS cold start and openPN issues #24 and #34

taivo committed:
format iOS payload to be consistent for openPN and receivePN. Related to #24, #27, and #34

taivo committed:
add example ionic app for faster dev and testing

taivo committed:
instructions to setup example app

taivo committed:
added sample parse-server and setup docs

taivo committed:
update docs

taivo committed:
update docs

taivo committed:
update docs

taivo committed:
move GCM_SENDER_ID into config.xml to simplify setup. No more plugin add --variable

taivo committed:
docs

benjie commented:
Closing this pull request since I no longer maintain this project.
